app/src/pages/auth/ResetPasswordWithTokenForm.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useCallback, useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+
+import {
+  Alert,
+  Button,
+  Flex,
+  Form,
+  TextField,
+  View,
+} from "@arizeai/components";
+
+const MIN_PASSWORD_LENGTH = 4;
+
+export type ResetPasswordWithTokenFormParams = {
+  resetToken: string;
+  newPassword: string;
+  confirmPassword: string;
+};
+
+interface ResetPasswordWithTokenFormProps {
+  resetToken: string;
+}
+
+export function ResetPasswordWithTokenForm({
+  resetToken,
+}: ResetPasswordWithTokenFormProps) {
+  const [message, setMessage] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const onSubmit = useCallback(
+    async ({ resetToken, newPassword }: ResetPasswordWithTokenFormParams) => {
+      setMessage(null);
+      setError(null);
+      setIsLoading(true);
+      try {
+        const response = await fetch("/auth/reset-password", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ token: resetToken, password: newPassword }),
+        });
+        if (!response.ok) {
+          setError("Failed attempt");
+          return;
+        }
+      } catch (error) {
+        setError("Failed attempt");
+        return;
+      } finally {
+        setIsLoading(() => false);
+      }
+      setMessage("Success");
```

Comment:
Not helpful message

Reply:
I would have something like 
<img width="606" alt="Screenshot 2024-09-17 at 8 36 20 AM" src="https://github.com/user-attachments/assets/3a6e2b15-668c-4048-9869-f0e5f10f147b">